libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -1926,27 +1928,31 @@ void prvGetLabel( CK_ATTRIBUTE_PTR * ppxLabel,
  * combination of the public and private key in DER format, and re-import of the
  * combination.
  */
-CK_RV prvGetExistingKeyComponent( CK_OBJECT_HANDLE_PTR pxPalHandle,
-                                  mbedtls_pk_context * pxMbedContext,
-                                  CK_ATTRIBUTE_PTR pxLabel )
+static CK_RV prvGetExistingKeyComponent( CK_OBJECT_HANDLE_PTR pxPalHandle,
+                                         mbedtls_pk_context * pxMbedContext,
+                                         CK_ATTRIBUTE_PTR pxLabel )
 {
     uint8_t * pucData = NULL;
     size_t xDataLength = 0;
     CK_BBOOL xIsPrivate = CK_TRUE;
-    BaseType_t xResult = CKR_OK;
-    int lMbedResult = 0;
+    CK_RV xResult = CKR_OK;
+    int32_t lMbedResult = 0;
 
     *pxPalHandle = CK_INVALID_HANDLE;
 
-    if( 0 == memcmp( pxLabel->pValue, pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS, pxLabel->ulValueLen ) )
+    if( 0 == strncmp( pxLabel->pValue, pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS, pxLabel->ulValueLen ) )
     {
         *pxPalHandle = PKCS11_PAL_FindObject( ( uint8_t * ) pkcs11configLABEL_DEVICE_PUBLIC_KEY_FOR_TLS, ( uint8_t ) pxLabel->ulValueLen );
     }
-    else if( 0 == memcmp( pxLabel->pValue, pkcs11configLABEL_DEVICE_PUBLIC_KEY_FOR_TLS, pxLabel->ulValueLen ) )
+    else if( 0 == strncmp( pxLabel->pValue, pkcs11configLABEL_DEVICE_PUBLIC_KEY_FOR_TLS, pxLabel->ulValueLen ) )
     {
         *pxPalHandle = PKCS11_PAL_FindObject( ( uint8_t * ) pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS, ( uint8_t ) pxLabel->ulValueLen );
         xIsPrivate = CK_FALSE;
     }
+    else
+    {
+        /* Unknown label passed to function */
```

Comment:
Shouldn't an error condition be set here then?

Reply:
The pxPalHandle is set to INVALID when initialized, so we will return an error